src/build_vehicle_gui.cpp

```diff
@@ -1212,7 +1212,7 @@ struct BuildVehicleWindow : Window {
 
 		this->details_height = ((this->vehicle_type == VEH_TRAIN) ? 10 : 9);
 
-		this->FinishInitNested(tile == INVALID_TILE ? (int)type : (int)tile);
+		this->FinishInitNested(tile == INVALID_TILE ? (int)type : tile);
```

Comment:
This looks odd, as the types of both sides are the same.

Reply:
Ugh, yes. We mix and matched this stuff way too much :D Lemme fix!